pkg/agents/team5/agent.go

```diff
@@ -86,6 +93,114 @@ func (a *CustomAgent5) checkForLeader() {
 	if diceRoll >= a.leadership {
 		a.Log("An agent has become a leader", infra.Fields{"dice roll": diceRoll, "leadership": a.leadership, "selfishness": a.selfishness, "floor": a.Floor()})
 		//TODO: Send treaties here about eating less food
+		hpLevel := a.currentAimHP - ((a.currentAimHP-a.HealthInfo().WeakLevel)/10)*(diceRoll-a.leadership)
+		a.currentProposal = messages.NewTreaty(messages.HP, hpLevel, messages.LeavePercentFood, 100, messages.GE, messages.EQ, 5, a.ID())
+		a.treatySendCounter = 1
+		a.Log("Agent is sending a treaty proposal", infra.Fields{"Proposed Max Hp": hpLevel})
+	}
+}
+
+func (a *CustomAgent5) statementsIntersect(op1 messages.Op, value1 int, op2 messages.Op, value2 int) bool {
+	switch op1 {
+	case messages.EQ:
+		return ((op2 == messages.EQ && value1 == value2) ||
+			(op2 == messages.LT && value1 < value2) ||
+			(op2 == messages.LE && value1 <= value2) ||
+			(op2 == messages.GT && value1 > value2) ||
+			(op2 == messages.GE && value1 >= value2))
+	case messages.LT:
+		return ((op2 == messages.EQ && value1 < value2) ||
+			op2 == messages.LT ||
+			op2 == messages.LE ||
+			(op2 == messages.GT && (value1-1) > value2) || // need the -1 here to disallow x<3 AND x>2
```

Comment:
If every condition value is an integer then we shouldn't need the -1 here. x<3 would be 0, 1, 2. x>2 would be 3, 4 ... Which don't intersect

Reply:
I think you want to use value1-2 > value2. If value1 is 4, and value2 is 2. These interesect at 3.

Reply:
yes these don't intersect, but would be marked as intersecting if the -1 was not there because for `x<3 AND x>2`,
`value1 = 3`, `value2 = 2`, returning `3 > 2 = true`
therefore I'm pretty sure we need the -1 to make `2 > 2 = false`
this is getting very confusing though aha